compiler/plugins/target/ROCM/ROCMTargetUtils.cpp

```diff
@@ -228,24 +184,66 @@ LogicalResult linkHIPBitcodeIfNeeded(Location loc, llvm::Module *module,
   return linkWithBitcodeFiles(loc, module, bitcodePaths);
 }
 
+static SmallVector<llvm::Expected<std::unique_ptr<llvm::Module>>>
+loadUKernelBitcodeFiles(llvm::LLVMContext &context, StringRef targetGpuArch) {
+  const iree_file_toc_t *toc = nullptr;
+  int toc_size = 0;
+  if (targetGpuArch == "gfx90a") {
+    toc = iree_uk_amdgpu_gfx90a_create();
+    toc_size = iree_uk_amdgpu_gfx90a_size();
+  } else if (targetGpuArch == "gfx942") {
+    toc = iree_uk_amdgpu_gfx942_create();
+    toc_size = iree_uk_amdgpu_gfx942_size();
+  } else if (targetGpuArch == "gfx1030") {
+    toc = iree_uk_amdgpu_gfx1030_create();
+    toc_size = iree_uk_amdgpu_gfx1030_size();
+  } else if (targetGpuArch == "gfx1100") {
+    toc = iree_uk_amdgpu_gfx1100_create();
+    toc_size = iree_uk_amdgpu_gfx1100_size();
+  } else {
+    return {};
+  }
```

Comment:
Similar comment here - could have the build system generate this if the list is expected to grow or be dynamic.

Or have a registration mechanism / dependency injection, so we don't have 5 places to update whenever adding a new architecture.

Reply:
See reply to previous comment. This code, currently in ROCMTargetUtils (ie linking-time) will move up to the earlier LowerToUkernels time where it will be key to how LowerToUkernels decides whether to succeed or fail. Separate arch-specific TOC's are going to be key.